src/indexer/merger.rs

```diff
@@ -103,204 +146,387 @@ impl IndexMerger {
     }
 
     fn write_fieldnorms(&self, fast_field_serializer: &mut FastFieldSerializer) -> Result<()> {
-        let fieldnorm_fastfields: Vec<Field> = self.schema
-            .fields()
-            .iter()
-            .enumerate()
-            .filter(|&(_, field_entry)| field_entry.is_indexed())
-            .map(|(field_id, _)| Field(field_id as u32))
-            .collect();
-        self.generic_write_fast_field(
-            fieldnorm_fastfields,
-            &extract_fieldnorm_reader,
-            fast_field_serializer,
-        )
+        for (field_id, field_entry) in self.schema.fields().iter().enumerate() {
+            if let FieldType::Str(ref text_options) = *field_entry.field_type() {
```

Comment:
Are fieldnorms only supported for string fields?

Reply:
yes... Rethinking that... Maybe it makes sense to enable them for ints as well. What do you think?

Reply:
You are right. There is no reason to not consider fieldnorm when scoring a query with `int fields`.

You actually have a small horse in this race, but its running the other way... You may end up having some index int fields, in which case field norms will just be a useless load for you (exactly one byte per doc in master). Note that this is also true for text fields so we probably want to make it configurable.